sql/group.go

```diff
@@ -181,6 +183,36 @@ func (n *groupNode) wrap(plan planNode) planNode {
 	return n
 }
 
+// desiredAggregateOrdering computes the desired output ordering from the
+// scan. It looks for an output column index containing a simle MIN/MAX
+// aggregation. If zero or multiple MIN/MAX aggregations are requested then no
+// ordering will be requested. A negative index indicates a MAX aggregation was
+// requested for the output column.
+func desiredAggregateOrdering(funcs []*aggregateFunc) []int {
+	var limit int
+	for i, f := range funcs {
+		switch f.impl.(type) {
+		case *maxAggregate, *minAggregate:
+			if limit != 0 || len(f.val.Exprs) != 1 {
+				return nil
+			}
+			switch f.val.Exprs[0].(type) {
+			case *qvalue:
+				limit = i + 1
+				if _, ok := f.impl.(*maxAggregate); ok {
+					limit = -limit
+				}
+			default:
+				return nil
+			}
+
+		default:
+			return nil
+		}
+	}
+	return []int{limit}
```

Comment:
Actually, I should be checking to see if `limit == 0` and returning an empty slice if it is. Will fix.